### PR TITLE
Generate serialization of CertificateInfo

### PR DESCRIPTION
**Pull request by achristensen07**
#### 370c4c4a9b951d20505cf6ef5fe2200e243f0a14
<pre>
Generate serialization of CertificateInfo
<a href="https://bugs.webkit.org/show_bug.cgi?id=248101">https://bugs.webkit.org/show_bug.cgi?id=248101</a>
rdar://102527605

Reviewed by Youenn Fablet.

* Source/WebCore/platform/WebCorePersistentCoders.cpp:
(WTF::Persistence::Coder&lt;WebCore::CertificateInfo&gt;::encode):
(WTF::Persistence::certificatesDataListFromCertificateInfo):
* Source/WebCore/platform/network/cf/CertificateInfo.h:
(WebCore::CertificateInfo::trust const):
* Source/WebCore/platform/network/cf/CertificateInfoCFNet.cpp:
(WebCore::CertificateInfo::containsNonRootSHA1SignedCertificate const):
* Source/WebCore/platform/network/cocoa/CertificateInfoCocoa.mm:
(WebCore::CertificateInfo::dump const):
* Source/WebCore/platform/network/soup/CertificateInfo.h:
(WebCore::CertificateInfo::certificate const):
* Source/WebCore/platform/network/soup/CertificateInfoSoup.cpp:
(WebCore::CertificateInfo::CertificateInfo):
* Source/WebCore/platform/network/soup/SoupNetworkSession.cpp:
(WebCore::SoupNetworkSession::allowSpecificHTTPSCertificateForHost):
* Source/WebCore/workers/service/server/SWServerJobQueue.cpp:
(WebCore::doCertificatesMatch):
* Source/WebKit/NetworkProcess/cocoa/NetworkProcessCocoa.mm:
(WebKit::NetworkProcess::allowSpecificHTTPSCertificateForHost):
* Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp:
(WebKit::NetworkDataTaskSoup::completeAuthentication):
* Source/WebKit/NetworkProcess/soup/WebSocketTaskSoup.cpp:
(WebKit::WebSocketTask::WebSocketTask):
* Source/WebKit/Platform/IPC/DaemonCoders.cpp:
(WebKit::Daemon::Coder&lt;WebCore::CertificateInfo&gt;::encode):
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/mac/WebCoreArgumentCodersMac.mm:
(IPC::ArgumentCoder&lt;WebCore::CertificateInfo&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;WebCore::CertificateInfo&gt;::decode): Deleted.
* Source/WebKit/Shared/soup/WebCoreArgumentCodersSoup.cpp:
(IPC::ArgumentCoder&lt;ResourceError&gt;::decodePlatformData):
* Source/WebKit/UIProcess/API/C/mac/WKFrameMac.cpp:
(WKFrameGetServerTrust):
* Source/WebKit/UIProcess/API/C/mac/WKPagePrivateMac.mm:
(-[WKObservablePageState serverTrust]):
* Source/WebKit/UIProcess/API/Cocoa/WKBrowsingContextController.mm:
(-[WKBrowsingContextController certificateChain]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView serverTrust]):
(-[WKWebView _certificateChain]):
(-[WKWebView certificateChain]):
* Source/WebKit/UIProcess/API/glib/WebKitCredential.cpp:
(webkit_credential_get_certificate):
* Source/WebKit/UIProcess/Inspector/mac/RemoteWebInspectorUIProxyMac.mm:
(WebKit::RemoteWebInspectorUIProxy::platformShowCertificate):
* Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm:
(WebKit::WebInspectorUIProxy::platformShowCertificate):
* Source/WebKit/WebProcess/InjectedBundle/API/Cocoa/WKWebProcessPlugInFrame.mm:
(-[WKWebProcessPlugInFrame _certificateChain]):
(-[WKWebProcessPlugInFrame _serverTrust]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebInspectorClient.mm:
(WebInspectorFrontendClient::showCertificate):

Canonical link: <a href="https://commits.webkit.org/257182@main">https://commits.webkit.org/257182@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/cb9ed1daaa06b4d12b9ae3236a5357b095c623fc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97997 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7217 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31158 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107464 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167738 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101935 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7694 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35987 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/90708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/104089 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103640 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5790 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84608 "Built successfully") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32945 "layout-tests (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87659 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89379 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1197 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20808 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1171 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22317 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6016 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44769 "Found 1 new test failure: fast/images/animated-heics-verify.html (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2464 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2462 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41719 "Passed tests") | |

### CODE REVIEW

File: Source/WebKit/Shared/glib/ArgumentCodersGLib.h [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #pragma once
 
 #include "ArgumentCoders.h"
+#include <gio/gio.h>
```
Comment (youennf) [inline Source/WebKit/Shared/glib/ArgumentCodersGLib.h:29]: Why is that needed?

**Review by youennf — APPROVED** [review]

File: Source/WebKit/Shared/glib/ArgumentCodersGLib.h (hunk 1 above)
Comment (mcatanzaro) [reply Source/WebKit/Shared/glib/ArgumentCodersGLib.h:29]: I suspect a forward declaration of ` enum GTlsCertificateFlags : int` would probably work instead? The actual declaration in C so it doesn't declare a storage class, but I guess `: int` corresponds to C enums.

`gio/gio.h` will pull in declarations for `GTlsCertificate` and `GVariant`, so those forward declarations can be removed if it is kept.

Comment (achristensen07) [reply Source/WebKit/Shared/glib/ArgumentCodersGLib.h:29]: I suspect that will make a symbol-redefined-as-different-type error since it's typedef typedef enum { ... } GTlsCertificateFlags;
I'm just going to do this at Zan's suggestion, and if you'd like to clean it up feel free.
